packages/core/src/modules/credentials/formats/indy/IndyCredentialFormatService.ts

```diff
@@ -0,0 +1,750 @@
+   * preview object as part of the stored credential record and need to add it to the
+   * proposal object used for processing credential proposals
+   * @param proposal the proposal object needed for acceptance processing
+   * @param preview the preview containing stored attributes
+   * @returns proposal object with extra preview attached
+   */
+  public setPreview(proposal: AcceptProposalOptions, preview: V2CredentialPreview): AcceptProposalOptions {
+    if (proposal.credentialFormats.indy) {
+      proposal.credentialFormats.indy.attributes = preview.attributes
+    }
+    return proposal
+  }
+
+  public async processProposal(
+    options: AcceptProposalOptions,
+    credentialRecord: CredentialExchangeRecord
+  ): Promise<AcceptProposalOptions> {
+    let proposalMessage
+    try {
+      proposalMessage = await this.didCommMessageRepository.getAgentMessage({
+        associatedRecordId: credentialRecord.id,
+        messageClass: V2ProposeCredentialMessage,
+      })
+    } catch (RecordNotFoundError) {
+      // can happen in normal processing
+    }
+    if (proposalMessage && proposalMessage.credentialProposal?.attributes) {
+      const proposeMessage = proposalMessage as V2ProposeCredentialMessage
+
+      let credPropose
+      const indyProposeFormat = proposeMessage.formats.find((f) => f.format.includes('indy'))
+
+      if (indyProposeFormat && proposeMessage.messageAttachment) {
+        const attachment = proposeMessage.messageAttachment.find(
+          (attachment) => attachment.id === indyProposeFormat.attachId
+        )
+        if (attachment) {
+          credPropose = this.getCredentialPayload<CredPropose>(attachment).indy?.payload
+            .credentialPayload as CredPropose
+        } else {
+          throw Error(`Missing data payload in attachment in credential Record ${credentialRecord.id}`)
+        }
+      }
+
+      if (proposeMessage && proposeMessage.credentialProposal) {
+        options.credentialFormats = {
+          indy: {
+            attributes: proposeMessage.credentialProposal.attributes,
+            credentialDefinitionId: credPropose?.credentialDefinitionId,
+          },
+        }
+        return options
+      }
+    }
+    throw Error('Unable to create accept proposal options object')
+  }
+
+  public getCredProposeOfferRequestFormat(message: RequestCredentialMessage | OfferCredentialMessage, id: string) {
+    const attachment = message.getAttachmentIncludingFormatId('indy')
+
+    if (attachment) {
+      return this.getCredentialPayload(attachment)
+    } else {
+      throw Error(`Missing (request) attachment in credential Record ${id}`)
+    }
+  }
+  /**
+   * Gets the attachment object for a given attachId. We need to get out the correct attachId for
+   * indy and then find the corresponding attachment (if there is one)
+   * @param message Gets the
+   * @returns The Attachment if found or undefined
+   */
+  public getAttachment(
+    message:
```

Comment:
This seems v2 specific. Maybe this can be moved to the v2 service and the format service will only include a method supportsFormatId. For all indy ones it will return true, but the v2credentialservice is still responsible for the extraction of the attachment (as it can change between versions, in v3 attachments will be placed in the appended/generic attachments array)